classytags/helpers.py

```diff
@@ -83,7 +84,10 @@ def render_tag(self, context, **kwargs):
             safe_context.update(**data)
             output = render_to_string(template, safe_context)
         else:
-            data = context.new(self.get_context(context, **kwargs))
+            new_context = context.new(
+                flatten_context(self.get_context(context, **kwargs))
+            )
+            data = flatten_context(new_context)
```

Comment:
This is the context fragment that generates the error: http://dpaste.com/0R5KE9Z (captured in a live session)